File: typescript/src/bot/bot.ts
import { PlayerAction } from "../models";
import { AbstractBot } from "./abstract-bot";
import { Room } from "colyseus.js";

export class Bot extends AbstractBot {
    constructor(game: Room) {
        super(game);
    }

    /** Insert the code for your bot here.  */
    doAction(): PlayerAction {
        return this.getRandomMove();
    }

    /** This method does not return plant bomb moves in order to protect the player. */
    getRandomMove() {
        const randomMove = Math.floor(Math.random() * 5);
        const action = new PlayerAction();

        if (randomMove === 0) {
            action.move_down = true;
        } else if (randomMove === 1) {
            action.move_up = true;
        } else if (randomMove === 2) {
            action.move_left = true;
        } else if (randomMove === 3) {
            action.move_right = true;
        }

        return action;
    }
}

File: python3/src/bot/bot.py
import math
from random import random
from bot.abstract_bot import AbstractBot
from models.player import player_action_factory

class Bot(AbstractBot):
    def __init__(self, game):
        super().__init__(game)

    # Insert the code for your bot here.
    def doAction(self):
        return self.getRandomMove()

    # This method does not return plant bomb moves in order to protect the player.
    def getRandomMove(self):
        randomMove = math.floor(random() * 5)
        action = player_action_factory()

        if randomMove == 0:
            action["move_down"] = True
        elif randomMove == 1:
            action["move_up"] = True
        elif randomMove == 2:
            action["move_left"] = True
        elif randomMove == 3:
            action["move_right"] = True

        return action
